### PR TITLE
Consolidating qn solver

### PR DESCRIPTION
**Pull request by cnpetra**
mostly improving documentation and codestyle for the linear algebra of the quasi-Newton solver.

### CODE REVIEW

**Comment by cnpetra** [comment]
@nychiang this would be a good moment to look into using CI to enforce coding style guidelines. For example, see @tepperly remark here https://github.com/LLNL/hiop/pull/693#discussion_r1751056011

File: src/Optimization/KktLinSysLowRank.hpp [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+// 3. Also, reference herein to any specific commercial products, process, or services by
+// trade name, trademark, manufacturer or otherwise does not necessarily constitute or
+// imply its endorsement, recommendation, or favoring by the United States Government or
+// Lawrence Livermore National Security, LLC. The views and opinions of authors expressed
+// herein do not necessarily state or reflect those of the United States Government or
+// Lawrence Livermore National Security, LLC, and shall not be used for advertising or
+// product endorsement purposes.
+
+/**
+ * @file KktLinSysLowRank.cpp
```
Comment (nychiang) [inline src/Optimization/KktLinSysLowRank.hpp]: cpp --> hpp

**Comment by nychiang** [comment]
@cameronrutherford would you mind checking the failed spack build? maybe we can setup a meeting and I can learn how to fix it from you.

File: src/Optimization/hiopIterate.hpp [hunk 2]
```diff
@@ -165,10 +172,9 @@ class hiopIterate
   friend class hiopKKTLinSysCompressedXDYcYd;
   friend class hiopKKTLinSysDenseXYcYd;
   friend class hiopKKTLinSysDenseXDYcYd;
-  friend class hiopKKTLinSysLowRank;
```
Comment (nychiang) [inline src/Optimization/hiopIterate.hpp:168]: do we have plan to drop the term 'hiop' from other classes/file names?

**Review by nychiang — APPROVED** [review]
LGTM. Will update BLT and check all the white spaces in another PR.

File: src/Optimization/hiopIterate.hpp (hunk 2 above)
Comment (cnpetra) [reply src/Optimization/hiopIterate.hpp:168]: yes for all new classes, for the rest will convert once we have a codestyle enforcer in place.

**Pull request merged**